test/visreg/tests/modals.ts

```diff
@@ -0,0 +1,62 @@
+import {By, Snappit, WebDriver, Key, WebElement, Browser} from "snappit-visual-regression";
+import {test, TestContext} from "ava";
+import * as util from "../../common/util";
+
+const getModal = async (
+    driver: WebDriver,
+    modalName: string
+) => {
+    if(modalName == "Open Modal") {
+        const modal = await util.$x(driver, util.selectors.openModal, modalName);
+        return modal;
+    }
+    else if(modalName == "Close Modal") {   
+        let modal = await util.$x(driver, util.selectors.openModal, "Open Modal");
+        await modal.click();
+        modal = await util.$x(driver, util.selectors.closeModal);
+        return modal;
+    }
+    else {
+        const modal = await util.$x(driver, util.selectors.modal, modalName);
+        return modal;
+    }
+}
```

Comment:
This abstraction can be handled directly in the test. The fact that you're passing in commands via `modalName` implies this.

Please handle the differences in opening and closing actions directly in the test body, instead of here in this abstraction.

Reply:
Thank you Andrew. Removed the entire modal.ts file